pkgs/development/compilers/futhark/default.nix

```diff
@@ -0,0 +1,27 @@
+{ callPackage, symlinkJoin, makeWrapper, stdenv, gcc, opencl-headers, opencl-icd }:
+
+let
+  cabal2nixPackage = callPackage ./package.nix {};
+  src = callPackage ./src.nix {};
+
+  package = cabal2nixPackage.overrideAttrs (drv: { inherit src; });
+
+  path = stdenv.lib.makeBinPath [ gcc ];
+
+  wrapped = symlinkJoin {
```

Comment:
Why do you use `symlinkJoin` with one one thing in `paths`?

Reply:
Makes it easy to patch a single file, since everything gets symlinked and I can just replace a symlink. Idris also uses this in its wrapper.

Alternative would be to do `ln -s ${package}/* $out; rm $out/bin; mkdir $out/bin; ln -s ${package}/bin/* $out/bin; rm $out/bin/futhark-c; rm $out/bin/futhark-opencl` and then the makeWrappers. I can do that if you prefer.